builtin/providers/ns1/permissions.go

```diff
@@ -0,0 +1,184 @@
+package ns1
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"gopkg.in/ns1/ns1-go.v2/rest/model/account"
+)
+
+func addPermsSchema(s map[string]*schema.Schema) map[string]*schema.Schema {
+	s["dns_view_zones"] = &schema.Schema{
+		Type:     schema.TypeBool,
+		Optional: true,
+	}
+	s["dns_manage_zones"] = &schema.Schema{
+		Type:     schema.TypeBool,
+		Optional: true,
+	}
+	s["dns_zones_allow_by_default"] = &schema.Schema{
```

Comment:
Hi @jbardin This comment is related to a failing acceptance test:

```
 $ make testacc TEST=./builtin/providers/ns1 TESTARGS='-run=TestAccTeam_updated'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/20 13:48:12 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/ns1 -v -run=TestAccTeam_updated -timeout 120m
=== RUN   TestAccTeam_updated
panic: empty structure "dns_zones_allow" has entry "dns_zones_allow_by_default"

goroutine 430 [running]:
panic(0x51e340, 0xc42048e2f0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/hashicorp/terraform/terraform.(*InstanceState).MergeDiff(0xc4202baeb0, 0xc4202d83e0, 0x0)
```

I could very easily be reading this wrong, but I believe that https://github.com/hashicorp/terraform/commit/2820845f8c3b21cb35666c29fa276da535ab99dd L1672 is supposed to check that the attribute is of either an array, map, or set?

The schema at fault in the ns1 provider has some subfields that share the same prefix(`dns_zones`):
 - `dns_zones_allow_by_default` as `schema.TypeBool`
 - `dns_zones_allow` as `schema.TypeList`
 - `dns_zones_deny` as `schema.TypeList`


I believe that the boolean `dns_zones_allow_by_default` shouldnt be included in that sanity check, am i thinking correctly on this?